frame/pablo/src/lib.rs

```diff
@@ -0,0 +1,782 @@
+			let pool = Self::get_pool(pool_id)?;
+			match pool {
+				PoolConfiguration::StableSwap(stable_swap_pool_info) =>
+					Ok(stable_swap_pool_info.pair),
+			}
+		}
+
+		fn get_exchange_value(
+			pool_id: Self::PoolId,
+			asset_id: Self::AssetId,
+			amount: Self::Balance,
+		) -> Result<Self::Balance, DispatchError> {
+			let pool = Self::get_pool(pool_id)?;
+			let pool_account = Self::account_id(&pool_id);
+			match pool {
+				PoolConfiguration::StableSwap(stable_swap_pool_info) =>
+					Self::get_exchange_value_for_stable_swap(
+						stable_swap_pool_info,
+						pool_account,
+						asset_id,
+						amount,
+					),
+			}
+		}
+
+		#[transactional]
+		fn add_liquidity(
+			who: &Self::AccountId,
+			pool_id: Self::PoolId,
+			base_amount: Self::Balance,
+			quote_amount: Self::Balance,
+			min_mint_amount: Self::Balance,
+			keep_alive: bool,
+		) -> Result<(), DispatchError> {
+			let pool = Self::get_pool(pool_id)?;
+			let pool_account = Self::account_id(&pool_id);
+			match pool {
+				PoolConfiguration::StableSwap(stable_swap_pool_info) => {
+					let mint_amount = Self::add_liquidity_for_stable_swap(
+						who,
+						stable_swap_pool_info,
+						pool_account,
+						base_amount,
+						quote_amount,
+						min_mint_amount,
+						keep_alive,
+					)?;
+					Self::deposit_event(Event::<T>::LiquidityAdded {
+						who: who.clone(),
+						pool_id,
+						base_amount,
+						quote_amount,
+						mint_amount,
+					});
+				},
+			}
+			Ok(())
+		}
+
+		#[transactional]
+		fn remove_liquidity(
+			who: &Self::AccountId,
+			pool_id: Self::PoolId,
+			lp_amount: Self::Balance,
+			min_base_amount: Self::Balance,
+			min_quote_amount: Self::Balance,
+		) -> Result<(), DispatchError> {
+			let pool = Self::get_pool(pool_id)?;
+			let pool_account = Self::account_id(&pool_id);
+			match pool {
+				PoolConfiguration::StableSwap(stable_swap_pool_info) => {
+					let (base_amount, quote_amount, updated_lp) =
+						Self::remove_liquidity_stable_swap(
+							who,
+							stable_swap_pool_info,
+							pool_account,
+							lp_amount,
+							min_base_amount,
+							min_quote_amount,
+						)?;
+					Self::deposit_event(Event::<T>::LiquidityRemoved {
+						pool_id,
+						who: who.clone(),
+						base_amount,
+						quote_amount,
+						total_issuance: updated_lp,
+					});
+				},
+			}
+			Ok(())
+		}
+
+		#[transactional]
+		fn exchange(
+			who: &Self::AccountId,
+			pool_id: Self::PoolId,
+			pair: CurrencyPair<Self::AssetId>,
+			quote_amount: Self::Balance,
+			min_receive: Self::Balance,
+			keep_alive: bool,
+		) -> Result<Self::Balance, DispatchError> {
+			let pool = Self::get_pool(pool_id)?;
+			let pool_account = Self::account_id(&pool_id);
+			match pool {
+				PoolConfiguration::StableSwap(pool) => {
+					// /!\ NOTE(hussein-aitlahcen): after this check, do not use pool.pair as the
+					// provided pair might have been swapped
+					ensure!(pair == pool.pair, Error::<T>::PairMismatch);
```

Comment:
We might not need these implementation details for now, as we can add them later with tests.